src/Atmos/Model/moisture.jl

```diff
@@ -239,3 +240,131 @@ function flux_diffusive!(moist::EquilMoist, flux::Grad, state::Vars, d_q_tot)
     flux.ρu += d_q_tot .* state.ρu'
     flux.moisture.ρq_tot += d_q_tot * state.ρ
 end
+"""
+  NonEquilMoist
+
+  Assumes that there isn't any thermodynamic equilibrium
+"""
+
+struct NonEquilMoist <: MoistureModel end
+
+vars_state(::NonEquilMoist, FT) = @vars(ρq_tot::FT, ρq_liq::FT, ρq_ice::FT)
+vars_gradient(::NonEquilMoist, FT) =
+    @vars(q_tot::FT, q_liq::FT, q_ice::FT, h_tot::FT)
+vars_diffusive(::NonEquilMoist, FT) = @vars(
+    ∇q_tot::SVector{3, FT},
+    ∇q_liq::SVector{3, FT},
+    ∇q_ice::SVector{3, FT}
+)
+vars_aux(::NonEquilMoist, FT) =
+    @vars(temperature::FT, θ_v::FT, q_liq::FT, q_ice::FT, p::FT, src_qliq::FT)
+
+@inline function atmos_nodal_update_aux!(
+    moist::NonEquilMoist,
+    atmos::AtmosModel,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+    TS = thermo_state(moist, atmos.orientation, state, aux)
+    aux.moisture.temperature = air_temperature(TS)
+    q_p_sat = PhasePartition_equil(
```

Comment:
Is this correct ? If we impose non-equilibrium thermodynamics here, is the `PhasePartition_equil` the right wrapper to use ? I recall seeing `NonEquil` or similar wrappers in `MoistThermodynamics.jl` and `states.jl`

Reply:
This should be correct since we are comparing the current thermodynamic state with what the state should be at equilibrium in order to calculate how much vapor becomes part of the cloud phase.

Reply:
I suppose the equil state is interpreted as a reference state here, as you mentioned

Reply:
Yes that is correct. The Equil calculation gives you the state to which you are relaxing to. But because of the assumed relaxation timescale you don't reach Equil state